src/lib/bootloader/systemdboot.rb

```diff
@@ -0,0 +1,304 @@
+      if status
+        _("enabled")
+      else
+        _("disabled")
+      end
+    end
+
+    # Secure boot setting shown in summary screen.
+    #
+    # @return [String]
+    def secure_boot_summary
+      link = if secure_boot
+        "<a href=\"disable_secure_boot\">(#{_("disable")})</a>"
+      else
+        "<a href=\"enable_secure_boot\">(#{_("enable")})</a>"
+      end
+
+      "#{_("Secure Boot:")} #{status_string(secure_boot)} #{link}"
+    end
+
+    # Display bootloader summary
+    # @return a list of summary lines
+    def summary(*)
+      result = [
+        Yast::Builtins.sformat(
+          _("Boot Loader Type: %1"),
+          "Systemd Boot"
+        )
+      ]
+      result << secure_boot_summary if Systeminfo.secure_boot_available?(name)
+      result
+    end
+
+    def name
+      "systemd-boot"
+    end
+
+    def packages
+      res = super
+
+      case Yast::Arch.architecture
+      when "x86_64", "aarch64"
+        res << "shim" << "mokutil" if secure_boot
+      else
+        log.warn "Unknown architecture #{Yast::Arch.architecture} for EFI"
+      end
+
+      res
+    end
+
+    def delete
+      delete_bootloader
+    end
+
+    # overwrite BootloaderBase version to save secure boot
+    def write_sysconfig(prewrite: false)
+      sysconfig = Bootloader::Sysconfig.new(bootloader: name,
+        secure_boot: secure_boot, trusted_boot: false,
+        update_nvram: false)
+      prewrite ? sysconfig.pre_write : sysconfig.write
+    end
+
+  private
+
+    LS = "/bin/ls"
+    KERNELINSTALL = "/usr/bin/kernel-install"
+    BOOTCTL = "/bin/bootctl"
+    CAT = "/bin/cat"
+    LOADERCONF = "/boot/efi/loader/loader.conf"
+    MOKUTIL = "/bin/mokutil"
+
+    def create_menue_entries
+      cmdline_file = File.join(Yast::Installation.destdir, "/etc/kernel/cmdline")
+      if Yast::Stage.initial
+        # kernel-install script needs the "root=<device>" entry in kernel parameters.
+        File.open(cmdline_file, "w+") do |fw|
+          fw.puts("root=#{Yast::BootStorage.root_partitions.first.name}")
+        end
+      end
+
+      Dir.foreach(File.join(Yast::Installation.destdir, "/usr/lib/modules")) do |kernel_name|
+        next if [".", ".."].include?(kernel_name)
+
+        kernel_file = File.join("/usr/lib/modules", kernel_name, "vmlinuz")
+        if File.exist?(File.join(Yast::Installation.destdir, kernel_file))
+          begin
+            Yast::Execute.on_target!(KERNELINSTALL, "add",
+              File.basename(kernel_name), kernel_file)
+          rescue Cheetah::ExecutionFailed => e
+            Yast::Report.Error(
+            format(_(
+                     "Cannot create systemd-boot menue entry:\n" \
+                     "Command `%{command}`.\n" \
+                     "Error output: %{stderr}"
+                   ), command: e.commands.inspect, stderr: e.stderr)
+          )
+            break
+          end
+        end
+      end
+      File.delete(cmdline_file) if Yast::Stage.initial
```

Comment:
why delete is needed? does not it prevent installation of params for kernel update? In code comment should be rationale

Reply:
Well, because it will be created here too. But maybe you are right and I should not delete it......

Reply:
I have checked it. /etc/kernel/cmdline will be used by the administrator in an installed system only. As we have created it in this function we should delete it again.